app/test/test_datastore.py

```diff
@@ -0,0 +1,383 @@
+            AttributeDefinitions=[{'AttributeName': 'PatientId', 'AttributeType': 'S'},],
+            ProvisionedThroughput={'ReadCapacityUnits': 5, 'WriteCapacityUnits': 5}
+        )
+        return table
+
+    @mock_dynamodb2 
+    def create_health_care_professional_table(self):
+        dynamodb = self.create_dyanamodb()
+        table = dynamodb.create_table(
+            TableName='HealthCareProfessionals',
+            KeySchema=[{'AttributeName': 'HealthCareProfessionalId', 'KeyType': 'HASH'},],
+            AttributeDefinitions=[{'AttributeName': 'HealthCareProfessionalId', 'AttributeType': 'S'},],
+            ProvisionedThroughput={'ReadCapacityUnits': 5, 'WriteCapacityUnits': 5}
+        )
+        return table
+
+    @mock_dynamodb2 
+    def create_session_table(self):
+        dynamodb = self.create_dyanamodb()
+        table = dynamodb.create_table(
+            TableName='Sessions',
+            KeySchema=[{'AttributeName': 'PatientId', 'KeyType': 'HASH'},{'AttributeName': 'SessionId', 'KeyType': 'RANGE'}],
+            AttributeDefinitions=[{'AttributeName': 'PatientId', 'AttributeType': 'S'},{'AttributeName': 'SessionId', 'AttributeType': 'S'}],
+            ProvisionedThroughput={'ReadCapacityUnits': 5, 'WriteCapacityUnits': 5}
+        )
+        return table
+
+    @mock_dynamodb2
+    def test_save_patient(self):
+        # test common save
+        tableName,patientId,patientName,patientIdVal,patientNameVal='Patients','PatientId','PatientName','p-1','Hello'
+        table = self.create_patient_table()
+        d = DataStore('dynamodb', tableName, patientId)
+        d.save({patientId: patientIdVal, patientName:patientNameVal},'us-west-2')
+        response = table.scan()
+        items = response['Items'] if 'Items' in response else {}
+        item = items[0] if items else None
+        self.assertTrue(patientIdVal in item[patientId])
+        self.assertEqual(item[patientName], patientNameVal)
+
+        # test save none value
+        patientIdVal, patientNameVal = 'p-2', None
+        response = d.save({patientId : patientIdVal, patientName : patientNameVal}, 'us-west-2')
+        self.assertEqual('PatientName should not be empty.', response['error'])
+        response = table.scan()
+        items = response['Items'] if 'Items' in response else {}
+        for item in items:
+            self.assertTrue(patientIdVal not in item[patientId])
+
+        # test save empty value
+        patientIdVal, patientNameVal = 'p-3', ''
+        response = d.save({patientId : patientIdVal, patientName : patientNameVal}, 'us-west-2')
+        self.assertEqual('PatientName should not be empty.', response['error'])
+        response = table.scan()
+        items = response['Items'] if 'Items' in response else {}
+        for item in items:
+            self.assertTrue(patientIdVal not in item[patientId])
+
+    @mock_dynamodb2
+    def test_save_session(self):
+         # test common save
+        tableName = 'Sessions'
```

Comment:
As discussed offline, The datastore is the lowest level of abstraction over which we build the models of Patient, session, HCP. Keeping that in mind, this test file should only check for datastore function : save, query by index etc and should not have to worry about save patient, save session etc. 
To test the higher level models and their functionalities, create a test_models.py file where you test the functions such as create_patient, list_patient etc

Reply:
Done.